mp_root_n.c

```diff
@@ -12,15 +12,18 @@
  * which will find the root in log(N) time where
  * each step involves a fair bit.
  */
-mp_err mp_root_u32(const mp_int *a, uint32_t b, mp_int *c)
+mp_err mp_root_n(const mp_int *a, int b, mp_int *c)
 {
    mp_int t1, t2, t3, a_;
-   mp_ord cmp;
    int    ilog2;
    mp_err err;
 
+   if (b < 0 || (unsigned)b > (unsigned)MP_DIGIT_MAX) {
```

Comment:
@czurnieden There is a check here.

Reply:
So you think I need a new prescription for my reading glasses? ;-)